components/create-bounty/token-amount/create-bounty-token-amount.tsx

```diff
@@ -0,0 +1,309 @@
+
+const ZeroNumberFormatValues = {
+  value: "",
+  formattedValue: "",
+  floatValue: 0,
+};
+
+interface DistributionsProps extends DistributedAmounts {
+  type: 'total' | 'reward';
+  totalServiceFees: BigNumber;
+}
+
+export default function CreateBountyTokenAmount({
+  currentToken,
+  setCurrentToken,
+  addToken,
+  canAddCustomToken,
+  defaultToken = null,
+  userAddress,
+  customTokens,
+  labelSelect,
+  tokenBalance,
+  issueAmount,
+  setIssueAmount,
+  isFunders = false,
+  needValueValidation,
+  decimals = 18,
+  isFunding = false,
+}) {
+  const { t } = useTranslation(["bounty", "common", "proposal"]);
+  const [show, setShow] = useState<boolean>(false);
+  const [rewardAmount, setRewardAmount] = useState<NumberFormatValues>(ZeroNumberFormatValues);
+  const [inputError, setInputError] = useState("");
+  const [distributions, setDistributions] = useState<DistributionsProps>();
+  const {
+    state: { currentUser, Service },
+  } = useAppState();
+
+  const debouncedDistributionsUpdater = useDebouncedCallback((value, type) => handleDistributions(value, type), 500);
+
+  function handleDistributions(value, type) {
+    if (!value || !Service?.network?.amounts) return;
+  
+    const { treasury, mergeCreatorFeeShare, proposerFeeShare } = Service.network.amounts;
+  
+    const distributions = calculateDistributedAmounts(treasury,
+                                                      mergeCreatorFeeShare,
+                                                      proposerFeeShare,
+                                                      BigNumber(value),
+                                                        [{recipient: currentUser?.walletAddress, percentage: 100}]);
+    
+    
+    
+    const totalServiceFees = distributions
+      ? [
+          distributions.mergerAmount.value,
+          distributions.proposerAmount.value,
+          distributions.treasuryAmount.value,
+      ].reduce((acc, value) => BigNumber(value).plus(acc),
+               BigNumber(0))
+      : BigNumber(0);
+
+    setDistributions({type, totalServiceFees, ...distributions})
+  }
+
+  useEffect(() => {
+    if(issueAmount?.value && !distributions){
+      debouncedDistributionsUpdater(issueAmount.value, 'total')
+    }
+  }, [issueAmount])
+
+  useEffect(() => {
+    if(distributions) {
+      if(distributions.type === 'reward'){
+        const total = distributions.totalServiceFees.plus(rewardAmount?.value) 
+        setIssueAmount({
+          value: total.toFixed(),
+          floatValue: total.toNumber(),
+          formattedValue: total.toFixed()
+        })
+      }
+
+      if(distributions.type === 'total'){
+        const rewardValue = BigNumber(issueAmount?.value).minus(distributions.totalServiceFees) 
+        setRewardAmount({
+          value: rewardValue.toFixed(),
+          floatValue: rewardValue.toNumber(),
+          formattedValue: rewardValue.toFixed()
+        })
+      }
+    }
+  }, [distributions])
```

Comment:
I think that both `useEffects` could be avoided if a function to handle distributions was called inside the `handleIssueAmountOnValueChange` function

Reply:
The second gives a state control that when the distribution is calculated it evaluates where it should be calculated and fills in the necessary input